docs/projects/tad/adrs/0005-ai-verify-technical-tests.md

```diff
@@ -0,0 +1,46 @@
+# TAD-0005 Use AI Verify for technical tests
+
+## Context
+
+The [AI Verify project](https://github.com/aiverify-foundation/aiverify) is set up in a modular way, and the
+technical tests is one of the modules. The AI Verify
+team is developing a feature which makes it possible to run the technical tests using an API, by using
+a Python file calling a method to run a test and providing the required configuration; for example, which
+model and dataset to use and some test specific configuration. The result of the test are returned in a JSON format,
+which can be processed in any way we please, like writing it to a file or System Card or store it in a database.
+
+### Pros
+
+* We have several technical tests we can use of the shelf, like SHAP, ALE, or Fairness metrics.
+* Tests are set up in a generic way using interfaces which allows others, like ourselves, to create their own plugins.
+* Loading models, pipelines and data is done through the AI Verify toolkit, which does validation
+ and support checks. All of these should adhere to a specific format, which one should be aware of.
+
+### Cons
+
+* The outcome of an AI Verify test depends on the implementation chosen by the tool's developers. This means
+ there's less control over what data goes into the process report compared to the final result report.
+* Adding our own plugins may require adding AI Verify frontend blocks we don't use ourselves.
```

Comment:
What do we mean bij this?

Reply:
Test from AI Verify consist of a "run the test" part (producing JSON), and "support and visualization" components (react components) for use in AI Verify's own UI tool, see https://aiverify-foundation.github.io/aiverify-developer-tools/plugins/widget/Plugin/.